### ISSUE TITLE
--cli flag doesn't work [Kubuntu 18.04.3]

### ISSUE THREAD

**Issue by Ocawesome101** [MCJack123/craftos2#53]
I run `craftos --cli` and it still opens a window. Running it from a TTY segfaults almost instantly.

Running `exit` (more specifically, calling `os.shutdown()`) after this results in the window hanging for about ten seconds (after I Ctrl-C it), then (maybe) segfaulting. See the screen recording in the zip I attached.
[2019-10-14 23-47-05.zip](https://github.com/MCJack123/craftos2/files/3727304/2019-10-14.23-47-05.zip)

Running without `--cli`, while it seems to work fine on my Ubuntu 19.04 (Unity) desktop computer, doesn't properly end the program. Again, see the screen recording.

In short, it doesn't quite work.

Oh, also, under KDE the side and bottom window borders don't quite render properly - they're white.

**Comment by MCJack123** [comment]
Could you run `about` in the shell? The bugs that you're seeing should have been fixed in 2.1, and I just want to rule out the possibility you're on an older version. Also, make sure `which craftos` points to the right executable (`/usr/bin/craftos`) and not some other path.

**Comment by Ocawesome101** [comment]
I’ll do that when I can get to my computer :) I installed from your PPA because that’s easier than building it, so that might be why I’ve got 2.0, if I’ve got 2.0. Thanks!

**Comment by Ocawesome101** [comment]
> Could you run `about` in the shell? The bugs that you're seeing should have been fixed in 2.1, and I just want to rule out the possibility you're on an older version. Also, make sure `which craftos` points to the right executable (`/usr/bin/craftos`) and not some other path.

On my desktop (19.04) I have 2.1 and cli mode works. However wget and gitget still do not.

On my laptop (18.04.3) I have 2.1 as well and cli mode does *not* work.

**Comment by Ocawesome101** [comment]
> Could you run `about` in the shell? The bugs that you're seeing should have been fixed in 2.1, and I just want to rule out the possibility you're on an older version. Also, make sure `which craftos` points to the right executable (`/usr/bin/craftos`) and not some other path.

Oh, also, after compiling from source `wget` and `gitget` both work :) however, my custom "os" (https://github.com/ocawesome101/oc-linux-cc) is extremely slow, and in fact crashes, when doing its boot logging (so printing to the screen probably needs to be optimized, both in CLI mode and not) while the same works flawlessly in CCEmuX.

In short, a fresh build pushed to your PPA would probably fix most of these issues.

Should I open a new issue for these things?

**Comment by MCJack123** [comment]
I've found the bug that you had with your OS: `os.clock` uses the C `clock()` function to determine the time, but 1) that gets the time since CraftOS-PC started, not the time since the computer started, and 2) it returns CPU time, which does not line up with real time when running multi-threaded.

I'm going to be releasing v2.1.1 to fix these issues soon, and the new version will be pushed to the PPA.

**Comment by Ocawesome101** [comment]
Thanks! :D

**Comment by MCJack123** [comment]
Also, found why the PPA version didn't have CLI support: I forgot to add ncurses as a dependency, meaning it was compiled without CLI support.

**Comment by Ocawesome101** [comment]
Oh cool. That might be an issue lol
